admin-ui/plugins/auth-server/redux/features/authNLdapSlice.js

```diff
@@ -0,0 +1,84 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+  ldapList: [],
+  loading: false,
+  saveOperationFlag: false,
+  errorInSaveOperationFlag: false,
+}
+
+const authNLdapSlice = createSlice({
+  name: 'authNLdap',
+  initialState,
+  reducers: {
+    getLdapList: (state) => {
+      state.loading = true
+    },
+    getLdapListSuccess: (state, action) => {
+      state.ldapList = action.payload || []
+      state.loading = false
+    },
+    getLdapListFailure: (state) => {
+      state.ldapList = []
+      state.loading = false
+    },
+    setCurrentItem: (state, action) => {
+      state.item = action.payload?.item || null
+    },
+    addLdap: (state) => {
+      state.loading = true
+      state.saveOperationFlag = false
+      state.errorInSaveOperationFlag = false
+    },
+    addLdapSuccess: (state) => {
+      state.loading = false
+      state.saveOperationFlag = true
+      state.errorInSaveOperationFlag = false
+    },
+    addLdapFailure: (state) => {
+      state.loading = false
+      state.saveOperationFlag = false
+      state.errorInSaveOperationFlag = true
+    },
```

Comment:
Consolidate save status flags

Two booleans (saveOperationFlag/errorInSaveOperationFlag) are hard to reason about; prefer a single finite-state string: 'idle' | 'pending' | 'succeeded' | 'failed'. This simplifies UI effects and avoids invalid combinations.


I can provide a patch to introduce saveStatus and migrate usages in components/sagas.


Also applies to: 43-57, 58-66